### PR TITLE
Fixes for merging nightly

### PR DESCRIPTION
**Pull request by Arceveti**

### CODE REVIEW

File: src/game/camera.c [hunk 1]
```diff
@@ -6172,17 +6158,16 @@ s32 rotate_camera_around_walls(UNUSED struct Camera *c, Vec3f cPos, s16 *avoidYa
...
                 if ((is_range_behind_surface(sMarioCamState->pos, cPos, wall, yawRange, SURFACE_WALL_MISC) == 0)
                     && (is_behind_surface(sMarioCamState->pos, wall))
                     // Also check if the wall is tall enough to cover Mario
                     && (!is_surf_within_bounding_box(wall, -1.f, 150.f, -1.f))) {
                     // Calculate the avoid direction. The function returns the opposite direction so add 180
                     // degrees.
-                    *avoidYaw = calc_avoid_yaw(yawFromMario, wallYaw) + DEGREES(180);
-                    camera_approach_s16_symmetric_bool(avoidYaw, horWallNorm, yawRange);
+                    *avoidYaw = calc_avoid_yaw(yawFromMario, wallYaw);
+                    camera_approach_s16_symmetric_bool(avoidYaw, wallYaw, yawRange);
```
Comment (thecozies) [inline src/game/camera.c:6170]: what are these changes?

File: src/game/object_helpers.c [hunk 2]
```diff
@@ -244,15 +244,15 @@ void obj_set_held_state(struct Object *obj, const BehaviorScript *heldBehavior)
...
 }
 
 f32 dist_between_objects(struct Object *obj1, struct Object *obj2) {
-    f32 dist;
-    vec3f_get_dist(&obj1->oPosVec, &obj2->oPosVec, &dist);
-    return dist;
+    register Vec3f d;
+    vec3_diff(d, &obj2->oPosVec, &obj1->oPosVec);
+    return vec3_mag(d);
 }
```
Comment (thecozies) [inline src/game/object_helpers.c:256]: performance tested on console?

**Review by thecozies — COMMENTED** [review]
couple questions

File: src/game/object_helpers.c (hunk 2 above)
Comment (Arceveti) [reply src/game/object_helpers.c:256]: This pretty much just reverts them to vanilla.

File: src/game/camera.c (hunk 1 above)
Comment (Arceveti) [reply src/game/camera.c:6170]: Fixing calc_avoid_yaw means wallYaw and horWallNorm end up being the same.

**Review by thecozies — APPROVED** [review]
###### _nice_

**Pull request merged**